content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,244 @@
+
+The above image shows the numbers of the points that mediapipe uses to refer to different points of the hand. This tutorial will use point `4` and point `8` which are the thumb and the index finger respectively.
+### **Creating a hand gesture volume controller**
+### Setting up
+First of all, we will prepare our workspace. Launch the Pycharm app. Click on the create a new project.
+
+![Pycharm](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/pycharm.png)
+
+Click on the create button on the window that appears next.
+
+We need to install the libraries we will use in our project.
+
+`numpy` will help us work with arrays. To install it, open the terminal and run the following command: 
+
+```bash
+pip install numpy
+```
+Repeat the same process for the other libraries.
+
+```bash
+pip install opencv-python
+```
+We will import this library as `cv2`. We will use it to capture an image using the webcam and convert it to `RGB`.
+
+```bash
+pip install mediapipe
+```
+It is an open-source library developed by Google. We use it for both face and gesture recognition. For this tutorial, we'll use it for hand gesture recognition.
+
+```bash
+pip install pycaw
+```
+We'll need this library to access the device's speaker and its master volume.
+
+```bash
+pip install python-math
+```
+We'll use this library to find the distance between point number `4` (the thumb) and point number `8` (the index finger) using hypotenuse.
+
+```bash
+pip install gpib-ctypes, comtypes
+```
+`pycaw` depends on these two libraries. `Ctypes` provides `C` language compatible data types. `Comtypes` bases on the `ctypes` **FFI**(Foreign Function Interface) library.
+
+Now, let's start coding. In the `main.py` file that *pycharm* automatically creates for you, type in the following code:
+
+### Step 1: Importing the libraries we will need
+```python
+import cv2
+import mediapipe as mp
+from math import hypot
+from ctypes import cast, POINTER
+from comtypes import CLSCTX_ALL
+from pycaw.pycaw import AudioUtilities, IAudioEndpointVolume
+import numpy as np
+```
+In the above code segment, we are importing each library we installed in our project.
+
+```python
+cap = cv2.VideoCapture(0) 
+```
+We then get the video input from our computer's primary camera. If you are using any other camera, replace the number `0` with that of the camera you are using.
+
+ Lets now detect, initialize and configure the hands in the video input.
+
+### Step 2: Detecting, initializing, and configuring the hands
+```python
+mpHands = mp.solutions.hands
+hands = mpHands.Hands()
+mpDraw = mp.solutions.drawing_utils
+```
+In the above code, we are calling on the `mediapipe` hand module to detect the hands from the video input we got from our primary camera. `MpHands.Hands()` then completes the initialization and configuration of the detected hands. We finally draw the *connections* and *landmarks* on the detected hand using `mp.solutions.drawing_utils`. 

```

Comment:
Please replace, 'In the above code', with 'In the code above'. It sounds more grammatically correct. Go through your entire tutorial and edit accordingly.